multi-arch/samples.md

```diff
@@ -0,0 +1,157 @@
+
+In particular:
+ - will support across the entire product be staged along homogoneous / heterogeneous lines
+ - if "homogeneous only" becomes a valid option for the released product, is use of Golang's APIs for determining architecture sufficient
+ - if "heterogenous" is an option, the assumption is the list of valid architectures will be provided in a cluster-wide API object, and that samples operator installs for all architectures listed
+ - if "heterogenous" is an option, some notion outside the scope of the cluster operator API determines if a given sample runs on its associated architecture (for example, is a node selector needed on a deployment config or build leveraging a non-x86 imagestream to make sure the execution occurs on the right architecture?)
+
+## Motivation
+
+Multi-arch has been declared a key initiative for v4.x.
+
+### Goals
+
+x86 samples are not installed in a cluster with no x86 nodes.  Nor are any future non-x86 samples installed in a cluster with only x86 nodes.
+
+### Non-Goals
+
+We do not take on porting existing samples to non-x86 architectures.
+
+## Proposal
+
+The samples operator config object already supports the specification of multiple 
+architectures.
+
+It currently requires that only a single architecture is specified, and that 
+it is x86_64.
+
+In conjunction, the import of content from https://github.com/openshift/library supports 
+downloading multiple architectures, and segregating multiple architectures.  And 
+the tagging in https://github.com/openshift/library allows providers to designate which
+samples go to which platforms.
+
+So enabling the above to allow PPC and Z content if it is available is item one.
+
+But as previously stated, there is currently no such content, so the samples operator
+should do nothing when it comes up in Managed state, and report Available=true, Progressing=false,
+Degraged=false.
+
+We also need to ensure that marking the samples operator as Unmanaged or Removed 
+
+There are some interesting details and caveats that will be discussed in the 
+implementation details section.
+
+### User Stories [optional]
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+In a somewhat related note, there have been a lot of side conversations regarding the Jenkins imagestream
+in particular, as the openshift/jenkins images are included in the 4.1/4.2 payloads.
+
+Clayton weighed in with Ben Parees and they concluded that unless/until the openshift/jenkins images 
+are fully removed from the install payload (i.e. the devtools team, say as part of the Jenkins operator work, 
+somehow facilitates the removal of the openshift/jenkins images from the payload), the x86 openshift/jenkins 
+images will be utilized as part of building the non-x86 install payload.
+
+### Risks and Mitigations
+
+Unexpected outcomes with respect to the overall multi-arch design.
+
+Unexpected pushback from IBM on the lack of non-x86 samples.
```

Comment:
nit: "lack of non-x86 samples" -> "ownership of non-x86 samples"?  To emphasize the fact that we all agree that we want them, so the pushback would probably be about who has to write them.  And maybe drop "from IBM", because non-OpenShift teams within Red Hat are potential candidates too.

Reply:
I would not categorize the risk as "pushback from IBM".  We want to support these platforms.  Suggest rephrasing as "Unexpected increased priority/requirement to include non-x86 samples"

Reply:
@bparees 's wording part of next push